packages/taquito/src/prepare/prepare-provider.ts

```diff
@@ -216,7 +216,7 @@ export class PrepareProvider extends Provider implements PreparationProvider {
         : await this.context.walletProvider.getPKH(),
       publicKey: isSignerConfigured
         ? await this.signer.publicKey()
-        : await this.context.walletProvider.getPK(),
+        : await this.context.wallet.pk(),
```

Comment:
good call to make the change can we also update line 216 to `: await this.context.wallet.pkh(),`